salt/_macros/kubectl.jinja

```diff
@@ -63,7 +63,25 @@
     - name:        {{ manifest }}
     - source:      {{ src }}
     - template:    jinja
+    - makedirs:    true
+  {%- if 'require' in kwargs %}
+    - require:
+    {%- for r in kwargs['require'] %}
+        - {{ r }}
+    {%- endfor %}
+  {%- endif %}
   {{ _kubectl_run("apply -f " + manifest,
                   watch=["file: " + manifest] + kwargs.pop('watch', []),
                   **kwargs) }}
-{%- endmacro %}
+    {#- make this idempotent by using a 'onchanges' on the file: #}
+    {#- we don't run 'kubectl apply' unless the file changes     #}
+    - onchanges:
+      - file: {{ manifest }}
+  {#- make sure we remove the manifest if something fails,     #}
+  {#- so we could run kubectl again if we orchestrate again... #}
+  cmd.run:      {#- cannot use file.absent to avoid duplicates #}
```

Comment:
I am curious why we can't write:
```
{{ manifest }}:
  file.absent
```

Reply:
Because there is a `file.managed` in the same block, and Salt does not allow two `file.*` statements in the same block... :-(

Reply:
Ok, thank you for explanation